includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -45,10 +59,11 @@ public static function instance() {
 	public function __construct() {
 		add_filter( 'manage_edit-job_listing_columns', [ $this, 'promoted_jobs_columns' ] );
 		add_action( 'manage_job_listing_posts_custom_column', [ $this, 'promoted_jobs_custom_columns' ], 2 );
+		add_action( 'admin_action_' . self::PROMOTE_JOB_ACTION, [ $this, 'handle_promote_job' ] );
+		add_action( 'admin_action_' . self::DEACTIVATE_PROMOTION_ACTION, [ $this, 'handle_deactivate_promotion' ] );
 		add_action( 'admin_footer', [ $this, 'promoted_jobs_admin_footer' ] );
-		add_action( 'load-edit.php', [ $this, 'handle_deactivate_promotion' ] );
 		add_action( 'wpjm_job_listing_bulk_actions', [ $this, 'add_action_notice' ] );
-
+		add_filter( 'allowed_redirect_hosts', [ $this, 'add_to_allowed_redirect_hosts' ] );
```

Comment:
I'm always adding `wpjobmanager.com` to the list of `allowed_redirect_hosts` here. I...don't think this is a problem, but this logic MIGHT be better suited in another place. WDYT?

Reply:
> I...don't think this is a problem

I agree that it's probably fine.

> this logic MIGHT be better suited in another place

I also agree with this. Since it's something very generic, maybe we could add to the `WP_Job_Manager` class? Checking the codebase, I couldn't find another place where it could fit.

Reply:
That might be a good idea! That, or maybe on `WP_Job_Manager_Com_API`, too.

I'll await on your input related [to this ](https://github.com/Automattic/WP-Job-Manager/pull/2481#discussion_r1256769424) to think the better way to handle this, too.